packages/next/src/server/app-render/csrf-protection.ts

```diff
@@ -0,0 +1,14 @@
+import { isMatch } from 'next/dist/compiled/micromatch'
+
+export const isCsrfOriginAllowed = (
+  originDomain: string,
+  allowedOrigins: string[] | undefined
```

Comment:
We should default the config to always have an allowedOrigins value, it should just be an empty array if it was omitted by the user. then we can get rid of some of the existential operators

Reply:
If that is too invasive a change for this PR then update the function to only accept an array and then move the existence check to the callsite

Reply:
The existence check in this case is to account for `TypeError: Expected pattern to be a non-empty string` which appears to be thrown at runtime when the user passes a blank string as an element of `allowedOrigins`

Reply:
What I was suggesting is that wherever the next.config.js is processed and turned into the value passed as `serverActions` we do a normalization step. We make allowedOrigins an array if it doesn't exist. we error if you provided a value but it was the incorrect format. We error if you provide an empty string etc...

We could even go as far as constructing the minimatch regex there so all we have to do is call the match here.

I just took a look though and it seems this logic is pretty spread out